src/cmd/list.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright © 2019 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/containers/toolbox/pkg/podman"
+	"github.com/containers/toolbox/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var listFlags struct {
+	listContainers bool
+	listImages     bool
+}
+
+var listCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List existing toolbox containers and images",
+	Run: func(cmd *cobra.Command, args []string) {
```

Comment:
We should switch to the `RunE` hook, and keep throwing the errors up the call stack instead of logging them immediately using `logrus`. That way, Cobra will present the error message in a consistent way to the user and we can use the right exit code.

Currently, what's happening is that some errors get logged twice, or errors from the command line parsing are presented differently than other kinds of errors and so on.

Reply:
Yes, you are right. `logrus` should only be used for logging debug information (with different levels of severity) but it shouldn't be used to control the flow of the program (methods `Fatal` and `Panic`).